hivemind/utils/asyncio.py

```diff
@@ -127,3 +127,14 @@ async def _put_items():
     finally:
         if not task.done():
             task.cancel()
+
+
+async def aiter_with_timeout(iterable: AsyncIterable[T], timeout: float) -> AsyncIterator[T]:
+    """Iterate over an async iterable, raise TimeoutError if another portion of data does not arrive within timeout"""
+    # based on https://stackoverflow.com/a/50245879
+    iterator = iterable.__aiter__()
+    while True:
+        try:
+            yield await asyncio.wait_for(iterator.__anext__(), timeout=timeout)
```

Comment:
Can you please add a test for this? I am not sure cancelling `__anext__()` works fine.

Example: create an async generator that sleeps for 0.2 seconds before yielding a value, then iterate it with timeouts 0.1 and 0.3.